### ISSUE TITLE
Module sometimes starts muted

### ISSUE THREAD

**Issue by fulhjort** [freesurfacemodules/FreeSurface#5]
It's hard to reproduce this bug, but sometimes the module starts muted and won't make any sound until you click the Mode selector a few times.

When I've intentionally tried to reproduce the bug it doesn't seem to work, but in patch I'm working on it started appearing after a while. I've checked the Rack log file but it contains nothing out of the ordinary.

**Comment by fulhjort** [comment]
https://github.com/user-attachments/assets/a9b4d4b6-78f9-4942-a62f-3bc07c012588

I was able to get a video of it. As you can see sound comes into the module, but nothing comes out until I click the Mode selector.

**Comment by freesurfacemodules** [comment]
To make it easier to narrow down, does this happen when the patch is first opened, when the module is first added to the patch, or at some time later after it has started running?

**Comment by fulhjort** [comment]
Usually I've noticed it once I open the patch. I _think_ it has happened maybe once or twice after the module has started running, but it was before the latest version, haven't used the current version enough to notice that happening though.

**Comment by freesurfacemodules** [comment]
I suspect I know what is causing this. I haven't been able to reproduce it on Linux, but bugs like this are platform dependent and appear intermittently, so that doesn't mean much. Try out the [v2.1.2 release here](https://github.com/freesurfacemodules/FreeSurface/releases). If that doesn't fix the issue on startup, let me know and I will investigate more deeply. If you're experiencing a similar issue after running the module for a while, that probably has a different cause which is more challenging to fix (numerical instability).

**Comment by fulhjort** [comment]
It didn't fix it completely.

I kept opening/closing Rack and sometimes it acted just like before, sometimes the left output from the module would send an incredibly high (or low) value and count down (or up - if the value was low) until it reached zero and then it would work.

![ermsg](https://github.com/user-attachments/assets/234858ce-df25-49c1-979f-a20c4ffe7eac)

**Comment by freesurfacemodules** [comment]
Interesting, only the left channel? Seems like one bug was disguising another bug. I will take another look.

**Comment by fulhjort** [comment]
Yeah, at least in these tests, but I always tend to use the left channel and rarely go the stereo route.

**Comment by freesurfacemodules** [comment]
OK, here's another attempt at a fix: [v2.1.3](https://github.com/freesurfacemodules/FreeSurface/releases). Thanks for bringing this to my attention by the way, I should have noticed this mistake a while ago.

**Comment by fulhjort** [comment]
Sadly doesn't solve the issue, as it still starts muted. But at least it doesn't appear to send an incredibly high/low value anymore.

**Comment by fulhjort** [comment]
Heh, in my previous attempt I started/closed Rack maybe 15-20 times and it didn't start with a high/low value from the module, but just now it did. So I'm not sure what to make of that.

**Comment by freesurfacemodules** [comment]
OK, in a few days I'll be back home with a Windows computer, which will make this process  more straightforward since I'll be able to test it directly. I'll work on this issue more at that time, so that it's not necessary to go back and forth many times to get this fixed.

**Comment by fulhjort** [comment]
Awesome! Don't get me wrong, it's a minor annoyance and not something that stops me using the module at all, I really do like it!

**Comment by freesurfacemodules** [comment]
Don't worry, I just didn't want to offload repetitively testing changes onto someone else. I've uploaded a new build, 2.1.4, and in my testing on Windows I haven't seen any issues on startup with this change.
